libs/agno/agno/os/client.py

```diff
@@ -0,0 +1,2185 @@
+        return headers
+
+    async def _get(self, endpoint: str) -> Any:
+        """Execute GET request.
+
+        Args:
+            endpoint: API endpoint path (without base URL)
+
+        Returns:
+            Parsed JSON response
+
+        Raises:
+            HTTPStatusError: On HTTP errors (4xx, 5xx)
+        """
+        if not self._http_client:
+            self._http_client = AsyncClient(timeout=self.timeout)
+
+        url = f"{self.base_url}{endpoint}"
+        headers = self._get_headers()
+
+        response = await self._http_client.get(url, headers=headers)
+        response.raise_for_status()
+        return response.json()
+
+    async def _post(self, endpoint: str, data: Optional[Dict[str, Any]] = None) -> Any:
+        """Execute POST request.
+
+        Args:
+            endpoint: API endpoint path (without base URL)
+            data: Request body data
+
+        Returns:
+            Parsed JSON response
+
+        Raises:
+            HTTPStatusError: On HTTP errors (4xx, 5xx)
+        """
+        if not self._http_client:
+            self._http_client = AsyncClient(timeout=self.timeout)
+
+        url = f"{self.base_url}{endpoint}"
+        headers = self._get_headers()
+
+        response = await self._http_client.post(url, json=data, headers=headers)
+        response.raise_for_status()
+        return response.json()
+
+    async def _post_form_data(self, endpoint: str, data: Dict[str, Any]) -> Any:
+        """Execute POST request with form data (for runs endpoints).
+
+        Args:
+            endpoint: API endpoint path (without base URL)
+            data: Form data dictionary
+
+        Returns:
+            Parsed JSON response
+
+        Raises:
+            HTTPStatusError: On HTTP errors (4xx, 5xx)
+        """
+        if not self._http_client:
+            self._http_client = AsyncClient(timeout=self.timeout)
+
+        url = f"{self.base_url}{endpoint}"
+        headers = self._get_headers()
+
+        response = await self._http_client.post(url, data=data, headers=headers)
+        response.raise_for_status()
+        return response.json()
+
+    async def _stream_post_form_data(self, endpoint: str, data: Dict[str, Any]) -> AsyncIterator[str]:
+        """Stream POST request with form data.
+
+        Args:
+            endpoint: API endpoint path (without base URL)
+            data: Form data dictionary
+
+        Yields:
+            str: Lines from the streaming response
+        """
+        if not self._http_client:
+            self._http_client = AsyncClient(timeout=self.timeout)
+
+        url = f"{self.base_url}{endpoint}"
+        headers = self._get_headers()
+
+        async with self._http_client.stream("POST", url, data=data, headers=headers) as response:
+            response.raise_for_status()
+            async for line in response.aiter_lines():
+                yield line
+
+    async def _post_empty(self, endpoint: str) -> bool:
```

Comment:
These methods look a bit too similar, we can probably simplify and just have get, post, delete?

Reply:
https://github.com/agno-agi/agno/pull/5613/commits/ddd1b3e5f00280430d3cc194c0fa3c0ef2182cab